Java/shared/src/main/java/com/lantanagroup/link/shared/auth/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,95 @@
+package com.lantanagroup.link.shared.auth;
+
+import com.azure.security.keyvault.secrets.SecretClient;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.security.SignatureException;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+import org.springframework.web.servlet.HandlerExceptionResolver;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+  private final HandlerExceptionResolver handlerExceptionResolver;
+  private final SecretClient secretClient;
+  private final JwtService jwtService;
+
+  public JwtAuthenticationFilter (SecretClient secretClient, JwtService jwtService, HandlerExceptionResolver handlerExceptionResolver) {
+    super();
+    this.secretClient = secretClient;
+    this.jwtService = jwtService;
+    this.handlerExceptionResolver = handlerExceptionResolver;
+  }
+
+  @Override
+  protected void doFilterInternal (HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
+
+    String secret = secretClient.getSecret("link-bearer-key").getValue();
```

Comment:
does this get called only once? If it will make a request to a vault service, such as key vault, each time it will be pretty slow.

Reply:
I cached it